src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TagHelperDescriptorCache.cs

```diff
@@ -17,5 +20,117 @@ internal static class TagHelperDescriptorCache
 
         internal static void Set(int hashCode, TagHelperDescriptor descriptor) =>
             s_cachedTagHelperDescriptors.Set(hashCode, descriptor);
+
+#nullable enable
+        internal static int GetTagHelperDescriptorCacheId(TagHelperDescriptor descriptor)
```

Comment:
This calculation is pulled from the current implementation of `GetHashCode()` in each of the respective types below, though I did a minor bit of refactoring on the for loops to make it a bit more readable.